### DISCUSSION TITLE
BFF Aggregation with Consul dynamic routes

### DISCUSSION THREAD

**Discussion by elenacalugaru** [ThreeMammals/Ocelot#1993] [General]
Hello,
Currently we use Ocelot as an Api gateway over multiple microservices. We also have some GET requests aggregates . Now we want to use Consul for service discovery with dynamic routes but we don't know if in this case the GET requests aggregates if they will be recognized. We saw on the docs that in order to have dynamic routes it is necessary to have 0 routes but for us it will be really important to also keep in place the aggregates. Is there any solution to accomplish this?

**Comment by raman-m** [comment]
Hi @elenacalugaru !
Welcome to Ocelot world! 🐯 

> Is there any solution to accomplish this?

To accomplish this you need 2 instances of Ocelot app
- 1st instance. Ocelot + Consul + [Dynamic Routing](https://ocelot.readthedocs.io/en/latest/features/servicediscovery.html#dynamic-routing)
- 2nd instance. Ocelot for BFF ([Aggregation](https://ocelot.readthedocs.io/en/latest/features/requestaggregation.html)) in front of the 1st instance.

Please, note! Having separated Backend for Frontend instance is best practice for microservices.

Hope it helps!

**Reply by elenacalugaru** [reply]
Hello @raman-m,
Thanks a lot for your response.
We already have a dedicated aggregation project and it is used mainly to handle thecases in which we couldn't use request aggregation with ocelot eg aggregation with POST.  So this means that the only solution that we have is to move the existing request aggregation with GET either in the existing aggregation proj or to create another ocelot project .

**Reply by raman-m** [reply]
>  we couldn't use request aggregation with ocelot eg aggregation with POST.

Oh, my gosh! It can be fixed by a life coding hack. It should not take more than one day of development.
But should we prioritize "any verb" for [Aggregation](https://ocelot.readthedocs.io/en/latest/features/requestaggregation.html) feature?
I know we have some other opened issues with the same problem.

> So this means that the only solution that we have is to move the existing request aggregation with GET either in the existing aggregation proj or to create another ocelot project .

Sad to hear this. But that's how it works.
